sql/core/src/main/scala/org/apache/spark/sql/catalog/interface.scala

```diff
@@ -26,6 +26,22 @@ import org.apache.spark.sql.catalyst.DefinedByConstructorParams
 // Note: all classes here are expected to be wrapped in Datasets and so must extend
 // DefinedByConstructorParams for the catalog to be able to create encoders for them.
 
+/**
+ * A catalog in Spark, as returned by the `listCatalogs` method defined in [[Catalog]].
+ *
+ * @param name name of the catalog
+ * @since 3.2.0
+ */
+class CatalogMetadata(
+    val name: String)
```

Comment:
if it's simply a string, shall we just let `listCatalogs` return `Dataset[String]`?

Reply:
If Spark only knows the catalog name, then probably `Dataset[String]` would be fine.

Reply:
ok let's keep it then, and add a `description` field so that it doesn't look so weird.

About the naming, how about just `Catalog`? to match `Database` and `Table` in this file.

Reply:
ah I see!